airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -241,8 +237,8 @@ def _read_incremental(
         has_slices = False
         for _slice in slices:
             has_slices = True
-            if self.should_log_slice_message(logger):
-                yield self._create_slice_log_message(_slice)
+            if self.get_slice_logger().should_log_slice_message(logger):
+                yield self.get_slice_logger().create_slice_log_message(_slice)
```

Comment:
Should we just always log and remove this `if`? It feels to me that if every time I call `create_slice_log_message`, I need to call `should_log_slice_message`, it's probably internal logic that is leaking. I think the interface of `create_slice_log_message` makes sense so we can use `DebugSliceLogger(debug_enabled)` or `DebugSliceLogger(log_level)` to instantiate the SliceLogger so that `manifest_declarative_source.py` can also re-use it. It would also remove `AlwaysLogSliceLogger`